### PR TITLE
Fix #3552 option_story_text_size.xml 

### LINKED ISSUE #3552: Merge option_story_text_size.xml into single xml file
Currently there are 3 versions of `option_story_text_size.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by shankarpriyank**
## Explanation
- Merged 2 files of  option_story_text_size.xml 
 - Fixes #3552

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing
all 3 files merged mobile
error ss
tablet

### CODE REVIEW

**Comment by shankarpriyank** [comment]
I did as instructed in Pr #4125 , the app in tablet crashes when clicked on the options button , and also the screen in mobile not works as intended 
And if the layout file for tablet is preserved the app works fine , and the screens also look as they are supposed to

**Comment by shankarpriyank** [comment]
PTAL  @bkaur-bkj

**Comment by rt4914** [comment]
> I did as instructed in Pr #4125 , the app in tablet crashes when clicked on the options button , and also the screen in mobile not works as intended And if the layout file for tablet is preserved the app works fine , and the screens also look as they are supposed to

@shankarpriyank Try these steps for tablet https://github.com/oppia/oppia-android/wiki/Frequent-Errors-and-Solutions#crashing-layout-tags-in-tablet

**Comment by shankarpriyank** [comment]
> > I did as instructed in Pr #4125 , the app in tablet crashes when clicked on the options button , and also the screen in mobile not works as intended And if the layout file for tablet is preserved the app works fine , and the screens also look as they are supposed to
> 
> @shankarpriyank Try these steps for tablet https://github.com/oppia/oppia-android/wiki/Frequent-Errors-and-Solutions#crashing-layout-tags-in-tablet

Well its working fine in tablet, but still the mobile screen in not working as intended
image

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 1]
```diff
@@ -423,4 +423,8 @@
   <!-- Hints and Solution Fragment -->
   <dimen name="hints_and_solution_recycler_view_padding_top">4dp</dimen>
   <dimen name="hints_and_solution_recycler_view_padding_bottom">112dp</dimen>
+
+  <!-- Options story text size -->
```
Comment (bkaur-bkj) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: @shankarpriyank this comment can be written as `Options Story Text Size` or `OptionsStoryTextSize` as this pattern is followed in these files

**Comment by bkaur-bkj** [comment]
> > > I did as instructed in Pr #4125 , the app in tablet crashes when clicked on the options button , and also the screen in mobile not works as intended And if the layout file for tablet is preserved the app works fine , and the screens also look as they are supposed to
> > 
> > 
> > @shankarpriyank Try these steps for tablet https://github.com/oppia/oppia-android/wiki/Frequent-Errors-and-Solutions#crashing-layout-tags-in-tablet
> 
> Well its working fine in tablet, but still the mobile screen in working as intended image
@shankarpriyank what issues are you facing in mobile? can you pls share or explain this here

**Comment by shankarpriyank** [comment]
hey @bkaur-bkj the mobile screen should look like this 
image
 <br>
And the current version looks like this
image

File: app/src/main/res/layout/option_story_text_size.xml [hunk 2]
```diff
@@ -13,12 +16,12 @@
     android:id="@+id/reading_text_size_item_layout"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:background="@drawable/general_item_background_border"
+    android:background="@{commonViewModel.selectedFragmentIndex == viewModel.itemIndex ? @drawable/general_item_background_border_cyan : @drawable/general_item_background_border}"
```
Comment (bkaur-bkj) [inline app/src/main/res/layout/option_story_text_size.xml]: @shankarpriyank actually problem here is that in tablet layout by default 1st element is selected, but that's not the same for mobile thus it shows unexpected results. I think we can just merge mobile and tablet layouts separately and not together. @rt4914 I am not able to figure out any possible solution for this case except above, can you pls suggest any.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@shankarpriyank @bkaur-bkj Thanks a lot. I have gone through the comments and UI and it does make sense to merge tablet files together and mobile files together. i.e. we cannot delete the tablet files because of the color issues.

Apologies if there has been any inconvenience. I did not consider the color case earlier.

**Comment by shankarpriyank** [comment]
> @shankarpriyank @bkaur-bkj Thanks a lot. I have gone through the comments and UI and it does make sense to merge tablet files together and mobile files together. i.e. we cannot delete the tablet files because of the color issues.
> 
> Apologies if there has been any inconvenience. I did not consider the color case earlier.

Hey @rt4914 no worries , I have made the changes

**Comment by shankarpriyank** [comment]
image
image
image
image

The ui now looks like this 

PTAL @rt4914

**Comment by shankarpriyank** [comment]
hi @bkaur-bkj I had a look at the failing tests and I found out I made a mistake while resolving the merge conflicts  PTAL @bkaur-bkj

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: @shankarpriyank Please make this change

**Review by rt4914 — CHANGES_REQUESTED** [review]
@shankarpriyank PTAL at this comment https://github.com/oppia/oppia-android/pull/4190/files#r806619620

**Comment by shankarpriyank** [comment]
As Requested
| Before  | After |
| --------- | -------- |
 mobile landscape | mobile landscape
mobile potrait | mobile potrait
tab potrait|tablet portrait
tab landscape|tablet landscape





I cannot figure out why the tablet layout is working as intended , I guess it was working fine till yesterday😅

Edit - The tablet screen is now working as intended 

| Before  | After |
| --------- | -------- |
tab landscape | image

**Comment by shankarpriyank** [comment]
> @shankarpriyank PTAL at this comment https://github.com/oppia/oppia-android/pull/4190/files#r806619620

Done

**Comment by shankarpriyank** [comment]
PTAL @rt4914

**Comment by bkaur-bkj** [comment]
@shankarpriyank even I am in doubt about why this is not working right, though it seems it should. PTAL @rt4914

**Comment by shankarpriyank** [comment]
@bkaur-bkj I resolved the merge conflicts and committed it , and and most of the tests are failing , I cannot trace the error , and the project running file on my system , except for that screen..

**Comment by shankarpriyank** [comment]
PTAL @bkaur-bkj

**Comment by bkaur-bkj** [comment]
@shankarpriyank all Bazel tests are failing here, You have not changed any dependency so they must have worked right, I am not sure why its failing

**Review by rt4914 — CHANGES_REQUESTED** [review]
Just merge with latest develop and the CI check should mostly pass.

**Comment by shankarpriyank** [comment]
> Just merge with latest develop and the CI check should mostly pass.

Done
The issue still persists

File: app/src/main/res/layout-sw600dp-port/option_story_text_size.xml [hunk 3]
```diff
@@ -20,9 +20,9 @@
     android:background="@{commonViewModel.selectedFragmentIndex == viewModel.itemIndex ? @drawable/general_item_background_border_cyan : @drawable/general_item_background_border}"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-port/option_story_text_size.xml]: This file should not be changed. 
1. Don't change anything in this file.
2. Shift this file back to `layout-sw600dp` folder.

File: app/src/main/res/values-land/dimens.xml [hunk 4]
```diff
@@ -450,8 +450,12 @@
...
+  
   <!-- OptionAudioLanguage -->
   <dimen name="option_audio_language_padding_start">36dp</dimen>
   <dimen name="option_audio_language_padding_end">36dp</dimen>
-</resources>
+
+  <!-- OptionsStoryTextSize -->
+  <dimen name="option_story_text_size_padding_start">36dp</dimen>
+  <dimen name="option_story_text_size_padding_end">36dp</dimen>
+  </resources>
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Please fix the indentation for last line. There should be no space at the start of resrouces.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 5]
```diff
@@ -445,5 +445,9 @@
...
   <!-- OptionAudioLanguage -->
   <dimen name="option_audio_language_padding_start">16dp</dimen>
-  <dimen name="option_audio_language_padding_end">16dp</dimen>
-</resources>
+  <dimen name="option_audio_language_padding_end">16dp</dimen>\
+  
+  <!-- OptionsStoryTextSize -->
+  <dimen name="option_story_text_size_padding_start">16dp</dimen>
+  <dimen name="option_story_text_size_padding_end">16dp</dimen>
+  </resources>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Please fix the indentation for last line. There should be no space at the start of resrouces.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 6]
```diff
@@ -445,5 +445,9 @@
 
   <!-- OptionAudioLanguage -->
   <dimen name="option_audio_language_padding_start">16dp</dimen>
-  <dimen name="option_audio_language_padding_end">16dp</dimen>
-</resources>
+  <dimen name="option_audio_language_padding_end">16dp</dimen>\
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: There is `\` at the end. This is the reason for all the failures as the app is unable to build.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 7]
```diff
@@ -441,8 +441,12 @@
...
+  
   <!-- OptionAudioLanguage -->
   <dimen name="option_audio_language_padding_start">16dp</dimen>
   <dimen name="option_audio_language_padding_end">16dp</dimen>
-</resources>
+
+  <!-- OptionsStoryTextSize -->
+  <dimen name="option_story_text_size_padding_start">16dp</dimen>
+  <dimen name="option_story_text_size_padding_end">16dp</dimen>
+  </resources>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Ditto. Remove initial space

File: app/src/main/res/values/dimens.xml [hunk 8]
```diff
@@ -651,8 +651,12 @@
...
+  
   <!-- OptionAudioLanguage -->
   <dimen name="option_audio_language_padding_start">16dp</dimen>
   <dimen name="option_audio_language_padding_end">16dp</dimen>
-</resources>
+  
+  <!-- OptionsStoryTextSize -->
+  <dimen name="option_story_text_size_padding_start">16dp</dimen>
+  <dimen name="option_story_text_size_padding_end">16dp</dimen>
+  </resources>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Ditto. Remove initial space

**Review by rt4914 — CHANGES_REQUESTED** [review]
@shankarpriyank PTAL

File: app/src/main/res/layout-sw600dp-port/option_story_text_size.xml (hunk 3 above)
Comment (shankarpriyank) [reply app/src/main/res/layout-sw600dp-port/option_story_text_size.xml]: @rt4914 If I revert changes here I will have to bring back the land file for the tablet too

**Comment by shankarpriyank** [comment]
PTAL @rt4914

File: app/src/main/res/values-land/dimens.xml (hunk 4 above)
Comment (shankarpriyank) [reply app/src/main/res/values-land/dimens.xml]: Done

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 5 above)
Comment (shankarpriyank) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: Done

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 6 above)
Comment (shankarpriyank) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: Sorry, I couldn't identify this earlier

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 7 above)
Comment (shankarpriyank) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: Done

File: app/src/main/res/values/dimens.xml (hunk 8 above)
Comment (shankarpriyank) [reply app/src/main/res/values/dimens.xml]: Done

**Review by bkaur-bkj — APPROVED** [review]
LGTM . PTAL @rt4914

**Comment by bhaktideshmukh** [comment]
@shankarpriyank Your branch has conflicts , try resolving them.
[https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github](url)
Other than that everthing looks fine to me.
Thanks.

**Comment by bhaktideshmukh** [comment]
@shankarpriyank 
Your branch has conflicts , try resolving them
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github
Thanks.

File: app/src/main/res/layout-sw600dp-port/option_story_text_size.xml (hunk 3 above)
Comment (rt4914) [reply app/src/main/res/layout-sw600dp-port/option_story_text_size.xml]: Why do you need to make changes for land file? 
Just for understanding
`sw600dp-land` -> tablet landscape file
`sw600dp-port` -> tablet portrait file
`sw600dp` -> tablet landscape + portrait file


And this PR is making a change such that the `sw600dp` is shifted to `sw600dp-port` which means we won't have landscape file for tablet.

**Comment by rt4914** [comment]
@shankarpriyank PTAL at https://github.com/oppia/oppia-android/pull/4190#discussion_r825769101

**Comment by shankarpriyank** [comment]
Hi @bhaktideshmukh  I have resolved the issues as requested by Rajat , but I am having a hard time resolving the merge conflict as it says 
image
And even when I pushed commits in the branch it did not ask me to resolve the conflicts

**Comment by bhaktideshmukh** [comment]
> Hi @bhaktideshmukh I have resolved the issues as requested by Rajat , but I am having a hard time resolving the merge conflict as it says image And even when I pushed commits in the branch it did not ask me to resolve the conflicts

As the image say the conflicts are too complex to be resolved using the web UI. Hence you need to resolve them using terminal. You can check [wiki](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#resolving-merge-conflicts-using-the-terminal) for more instructions.

**Comment by shankarpriyank** [comment]
@bhaktideshmukh  Can you  please connect over a call  actually this has happened once before too, while trying to resolve the conflicts I ended up messing up the branch like I guess I have done here

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh Can you please connect over a call actually this has happened once before too, while trying to resolve the conflicts I ended up messing up the branch like I guess I have done here

It will be hard to connect over a call.
Did you tried running the commands from wiki of resolving merge conflicts which I mentioned in the last comment.
If not then , do try running them and please paste the output. I will provide you the direction.

**Comment by shankarpriyank** [comment]
@bhaktideshmukh I tried running the commands and ended up messing up this branch, I guess I will have to force push now to clean it up, so it's better if I close this and make a new clean PR

**Pull request closed without merging**